adam-core/src/main/scala/org/bdgenomics/adam/rdd/ADAMContext.scala

```diff
@@ -1562,11 +1562,29 @@ class ADAMContext(@transient val sc: SparkContext) extends Serializable with Log
     stringency: ValidationStringency = ValidationStringency.STRICT)(implicit s: DummyImplicit): AlignmentRecordRDD = LoadIndexedBam.time {
 
     val path = new Path(pathName)
+
+    // If pathName is a directory of bam files, scan all files to
+    // select index files.
+    val indexPath = if (pathName.endsWith(".bam"))
+      new Path(pathName + ".bai")
+    else
+      path
+
     // todo: can this method handle SAM and CRAM, or just BAM?
     val bamFiles = getFsAndFiles(path).filter(p => p.toString.endsWith(".bam"))
 
+    val indexFiles = getFsAndFiles(indexPath).filter(p => p.toString.endsWith(".bai"))
+      .map(r => r.toString)
+
     require(bamFiles.nonEmpty,
       "Did not find any BAM files at %s.".format(path))
+
+    // Make sure each bam file has an index
+    bamFiles.foreach(file => {
```

Comment:
Nit: if multiple files are missing an index, this will just throw for the first one. Instead, I'd suggest:

```
val missingIndices = bamFiles.filter(f => indexFiles.contains(f.toString + ".bai"))
require(missingIndices.isEmpty,
  "Missing indices for BAMs:\n%s".format(missingIndices.mkString("\n"))
```